.github/workflows/deprecation-lint.yml

```diff
@@ -0,0 +1,31 @@
+name: Deprecation ESLint check
+
+on:
+  workflow_call:
+  pull_request:
+    types: [opened, synchronize]
+    branches-ignore: [staging, production]
+    paths: ['**.js', '**.ts', '**.tsx', '**.json', '**.mjs', '**.cjs', 'config/.editorconfig', '.watchmanconfig', '.imgbotconfig']
+
+concurrency:
+  group: ${{ github.ref == 'refs/heads/main' && format('{0}-{1}', github.ref, github.sha) || github.ref }}-deprecation-lint
+  cancel-in-progress: true
+
+jobs:
+  deprecation-lint:
+    name: Deprecation ESLint check
+    if: ${{ github.actor != 'OSBotify' || github.event_name == 'workflow_call' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: ./.github/actions/composite/setupNode
+      
+      - name: Run ESLint to check for deprecation warnings
+        run: |
+          git fetch origin main --no-tags --depth=1
```

Comment:
What is the point of this fetch?

Reply:
oh, nevermind I see it: https://github.com/Expensify/App/actions/runs/10846117016/job/30098408343?pr=49007

Maybe just add a comment that we fetch main so we can check what files changed between the current ref and main